test/widgets/message_list_test.dart

```diff
@@ -605,6 +606,86 @@ void main() {
           expectedMessage: 'NetworkException: Oops (ClientException: Oops)');
       });
     });
+
+    group('Update Narrow on message move', () {
+      final channel = eg.stream(name: 'move test stream');
+      final message = eg.streamMessage(stream: channel, content: 'Message to move');
+      final narrow = TopicNarrow.ofMessage(message);
+
+      void prepareGetMessageResponse(List<Message> messages) {
+        connection.prepare(json: newestResult(
+          foundOldest: false, messages: messages).toJson());
+      }
+
+      void handleMessageMoveEvent(List<StreamMessage> messages, String newTopic) {
+        store.handleEvent(eg.updateMessageEventMoveFrom(
+          origMessages: messages,
+          newTopic: newTopic,
+          propagateMode: PropagateMode.changeAll));
+      }
+
+      testWidgets('compose box send message after move', (WidgetTester tester) async {
```

Comment:
Let's have this test move the message to a new channel as well as topic. That way we more fully exercise things changing.